src/User/Controller/SettingsController.php

```diff
@@ -539,4 +558,39 @@ protected function disconnectSocialNetwork($id)
         $account->delete();
         $this->trigger(SocialNetworkConnectEvent::EVENT_AFTER_DISCONNECT, $event);
     }
+
+    public function actionTwoFactorMobilePhone($id)
+    {
+        Yii::$app->response->format = Response::FORMAT_JSON;
+
+        /**
+        * 
+        *
+        * @var User $user 
+        */
+        $user = $this->userQuery->whereId($id)->one();
+
+        if (null === $user) {
+            return [
+                'success' => false,
+                'message' => Yii::t('usuario', 'User not found.')
+            ];
+        }
+        $mobilePhone = Yii::$app->request->get('mobilephone');
+        $currentMobilePhone = $user->getAuthTfMobilePhone();
+        $success=false;
+        if($currentMobilePhone==$mobilePhone){
+            $success=true;    
+        }else{
+            $success = $user->updateAttributes(['auth_tf_mobile_phone' => $mobilePhone]);
+            $this->make(TwoFactorSmsCodeGeneratorService::class, [$user])->run();
```

Comment:
If the procedure has been successful must depend on if the text message has been sent or not also, otherwise a "flashMessage" is set but it is not displayed since the page is not reloaded